modules/transactions.js

```diff
@@ -649,17 +657,25 @@ Transactions.prototype.shared = {
 		async.waterfall(
 			[
 				function getConfirmedCountFromCache(waterCb) {
-					modules.cache.getJsonForKey(
-						modules.cache.KEYS.transactionCount,
-						(err, data) => {
-							if (err) {
-								// If some issue in cache we will fallback to database
-								return setImmediate(waterCb, null, null);
+					if (library.config.cacheEnabled) {
```

Comment:
I would suggest to do `if (components.cache)` and initialise `components.cache` inside `app.js` if `library.config.cacheEnabled`